### PR TITLE
Tso500 ctdna full pipeline

### PR DESCRIPTION
**Pull request by alexiswl**
Added in the coupling workflow -> That is the glue between the [cwl tso500 ctDNA pipeline](https://github.com/umccr/cwl-ica/blob/main/workflows/tso500-ctdna/1.1.0--120/tso500-ctdna__1.1.0--120.cwl) and the [cwl tso500 ctDNA post processing pipeline](https://github.com/umccr/cwl-ica/blob/main/workflows/tso500-ctdna-post-processing-pipeline/1.0.0/tso500-ctdna-post-processing-pipeline__1.0.0.cwl)

### 5ded20490bb393e5ffd7a91e6770e601d7858490

First commit of the coupling workflow:

##### Pre ctDNA TSO500 workflow steps
  * get_tso_manifest_bed_from_resources_dir -> Need bed file for post processing workflow, user only puts in `resources` directory as an input into the workflow though.  
  * parse_tso_manifest_file -> Way to break down an array of files to a single file.
  
##### ctDNA TSO500 workflow step
  * Call the ctDNA TSO500 workflow step
  
##### Post processing step
  * No steps in between the workflow and the post processing step 
  * This one is scattered out for each sample in the workflow 

### 08d1a97c4a477ca6ac16f3e7439e27caeb29ce3b

Appended creation of the final directory structure to the workflow

#### Bug updates
* Stitched bam should be cleaned.stitched.bam and from the variant caller directory not the stitched realigned directory

##### Pre step to create final directory step:
* Pre-step in the full coupling workflow to create the final output directory step

##### Create final directory step:
* Create a final output directory containing the metrics output tsv file, the dsdm json file along with the sample post-processing directories

#### Bug updates
* Small bug fix to fix up the input parsers for the custom-output-directory tool

#### Bug updates
* Moved from hints to requirements for within steps to overcome https://github.com/common-workflow-language/cwltool/issues/1484

### 67b4a0e61d3ebe89434f08a4432952fe20203726

* Added expression 'get-file-from-directory' (single version of get-files-from-directory)
* Use 'get-file-from-directory' single rather than array based 'get-files-from-directory'

* Added expression to get a subdirectory from a directory (with deep listing)
* Added expression to 'get a string attribute from the tso50 sample object'
* Added step to get the sample output directory as an output to the full workflow

* Added 'parse-string' expressions as extra goodies

### a2939da40749c3278c109154472750419a5374ab

* Added seraseq run to tso500 ctdna pipeline

### 03d9cf8daa02972589d4bdf224960d71dc46b993

* Added workflow to production

### CODE REVIEW

**Comment by alexiswl** [comment]
@skanwal this one is ready to go just need an approval before I can merge it

**Comment by skanwal** [comment]
@alexiswl - I was going to ask you if this is ready as new few new commits.
Going through it now for my info.

**Review by skanwal — APPROVED** [review]

**Pull request merged**
